File: backend/src/cart/cart.service.ts
import { Injectable } from '@nestjs/common';
import { Product } from '../product/product.model';
import { Cart } from './cart.model';

@Injectable()
export class CartService {
  private cart: Cart[] = [];

  addToCart(cartId: number, product: Product): void {
    let existingCart = this.cart.find(cart => cart.id === cartId);

    if (!existingCart) {
      existingCart = {
        id: cartId,
        totalPrice: 0,
        productList: [],
      };
      this.cart.push(existingCart);
    }

    existingCart.productList.push(product);
    existingCart.totalPrice += product.price;
  }

  getCart(cartId: number): Cart {
    return this.cart.find(cart => cart.id === cartId);
  }
}

File: shop/lib/servies/cart_service.dart
import 'package:shop/models/cart_model.dart';
import 'package:shop/models/product_model.dart';

class CartService {
  Cart _cart = Cart(id: 1, productList: [], totalPrice: 0.0);

  void addToCart(Product product) {
    _cart.productList.add(product);
    _cart.totalPrice += product.price;
  }

  Cart getCart() {
    return _cart;
  }

  void clearCart() {
    _cart = Cart(id: 1, productList: [], totalPrice: 0.0);
  }
}
